lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -159,6 +159,15 @@ struct StructuralHasher {
 
   void update(OpResult result) {
     record(result.getAsOpaquePointer());
+
+    // Like instance ops, don't use object ops' result types since they might be
+    // replaced by dedup. Record the class names and lazily combine their hashes
+    // using the same mechanism as instances and modules.
+    if (auto objectOp = dyn_cast<ObjectOp>(result.getOwner())) {
+      referredModuleNames.push_back(objectOp.getType().getNameAttr().getAttr());
+      return;
+    }
```

Comment:
I think this part should be moved to `update(Type type)` to properly handle class types on ports.